### PR TITLE
[7.x-3.x] add in TFA methods

### PR DESCRIPTION
**Pull request by tobybellwood**
This includes

- tfa module
- tfa_basic plugins
- Patch to tfa_basic (https://www.drupal.org/project/tfa_basic/issues/2090033) to provide email as a TFA option - needed for agencies without access to mobile phones.

### CODE REVIEW

**Comment by tobybellwood** [comment]
Test this

File: drupal-org.make [hunk 1]
```diff
@@ -147,6 +147,9 @@ projects[strongarm][version] = "2.0"
 projects[swiftmailer][version] = "1.7"
 projects[superfish][version] = "1.9"
 projects[taxonomy_manager][version] = "1.0"
+projects[tfa][version] = "2.0"
```
Comment (ruwanl) [inline drupal-org.make]: projects[tfa][patch][] = "https://www.drupal.org/files/issues/2019-02-07/tfa-2820710-42.patch"

Comment (ruwanl) [inline drupal-org.make]: projects[tfa][patch][] = "https://www.drupal.org/files/issues/2019-02-07/tfa-2820710-42.patch"

**Review by ruwanl — CHANGES_REQUESTED** [review]
Can the patch be added to fix the issue at https://www.drupal.org/project/tfa/issues/2820710
projects[tfa][patch][] = "https://www.drupal.org/files/issues/2019-02-07/tfa-2820710-42.patch"

**Review by ruwanl — APPROVED** [review]

**Comment by tobybellwood** [comment]
I've updated the email patch on this to the most recent available - this may provide a better outcome, it appears to be more PHP7.1 compatible.  If we can't get it to work, we will remove the email patch for now and proceed with TOTP-only auth whilst we work on other methods.

**Review by pandaskii — APPROVED** [review]
PR and two patches look good

Email patch - Drupal PHPUnit test failed for some reason but could be a false alarm.

We can cover the test case for email as available two-factor authentication method during the QA.

**Pull request merged**